### PR TITLE
Revision of Execute Line in Object Attributes

### LINKED ISSUE #226: JIP ID is not received in Achilles_fnc_spawn
**Arma 3 Version:** `1.80` (stable)  
**CBA Version:** `3.5.0` (stable)  
**AresModAchillesExpansion Version:** `1.0.0` (stable)  

**Mods:**
```
- CBA_A3
- AresModAchillesExpansion
```

**Description:**
- Achilles_fnc_spawn should return a JIP ID number but fails to do so, it, in turn, creates a message in the .rpt log and returns an empty string as the JIP ID.

**Steps to reproduce:**
- Use any modules that use Achilles_fnc_spawn, e.g. the Lamps module etc.
- Observe in the Edit panel that there is an Entry for "JIP ID: <blank>"
- Notice warning message in .rpt.

**Where did the issue occur?**
- Dedicated

**RPT log file:**
```
Trying to call remoteExec(Call) with a null-object as JIP ID
```

### PR DESCRIPTION
**Pull request by oOKexOo**
**When merged this pull request will:**
- Replace the execute line by the new vehicle init line (object attribute window).
    Differences:
    - The code is executed globally and for JIPs.
- Fix #226.
- Make the vehicle init line available when the execute module is available.

### CODE REVIEW

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_spawn.sqf [hunk 1]
```diff
@@ -16,9 +16,33 @@
...
 
-params ["_args", ["_code", {}, [{}]], ["_target", clientOwner, [0, [], objNull, grpNull]], ["_jip", false, [false, "", objNull]]];
+params ["_args", ["_code", {}, [{}]], ["_target", 0, [0, [], objNull, grpNull, sideUnknown, ""]], ["_jip", false, [false, "", objNull]]];
 
-private _jip_id = "";
+private _jipId = false;
+
+if (isMultiplayer) then
+{
+	switch (typeName _jip) do
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_spawn.sqf]: isEqualType?

File: @AresModAchillesExpansion/addons/ui_f/scripts/RscAttributeExec.sqf [hunk 2]
```diff
@@ -0,0 +1,58 @@
+#include "\A3\ui_f_curator\ui\defineResinclDesign.inc"
+
+_mode = _this select 0;
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/ui_f/scripts/RscAttributeExec.sqf]: params

File: @AresModAchillesExpansion/addons/ui_f/scripts/RscDisplayAttributes.sqf [hunk 3]
```diff
@@ -57,16 +57,21 @@ switch _mode do
 		{
 			_enableAdmin = (_enableDebugConsole == 1 && (call BIS_fnc_admin > 0 || isServer)) || !isMultiplayer || _enableDebugConsole == 2;
 		};
+
+		// mission designer can disallow usage of code execution, but it will still be available for logged-in admins
+		private _enableCodeExecution = ((missionNamespace getVariable ['Ares_Allow_Zeus_To_Execute_Code', true]) or (serverCommandAvailable "#kick"));
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/ui_f/scripts/RscDisplayAttributes.sqf]: bis_fnc_admin?

**Comment by CreepPork** [comment]
Pretty sure that the variable this worked previously too.

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_spawn.sqf (hunk 1 above)
Comment (oOKexOo) [reply @AresModAchillesExpansion/addons/functions_f_achilles/functions/common/fn_spawn.sqf]: how does this command fit in here?

File: @AresModAchillesExpansion/addons/ui_f/scripts/RscAttributeExec.sqf (hunk 2 above)
Comment (oOKexOo) [reply @AresModAchillesExpansion/addons/ui_f/scripts/RscAttributeExec.sqf]: Agreed. Btw. that was bohemia, not me :D

**Comment by oOKexOo** [comment]
Yes, you are right. It already works in vanilla. I forgot it is a global variable.

**Comment by oOKexOo** [comment]
I addressed the latter two points.

**Pull request merged**